### PR TITLE
SNOW-706842 add a parameter if_not_exists

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-706842

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Adding a parameter `if_not_exists` in sp/udf/udtf registration so that users can skip creating procedure or function in the case when one is already registered.

### CODE REVIEW

**Review by sfc-gh-jdu — APPROVED** [review]

File: src/snowflake/snowpark/functions.py [hunk 1]
```diff
@@ -3136,6 +3137,10 @@ def udf(
             If it is ``False``, attempting to register a UDF with a name that already exists
             results in a ``SnowparkSQLException`` exception being thrown. If it is ``True``,
             an existing UDF with the same name is overwritten.
+        if_not_exists: Whether to skip creation of a UDF when one with the same signature already exists.
+            The default is ``False``. ``if_not_exists`` and ``replace`` are used mutually exclusive
+            and a ``SnowparkSQLException`` is raised when both are set. If it is ``True`` and a UDF with
```
[the remark this answers is not in the document]
Comment (sfc-gh-sfan) [reply src/snowflake/snowpark/functions.py:3142]: +1. Also make sure this case is tested.

File: src/snowflake/snowpark/functions.py [hunk 2]
```diff
@@ -3136,6 +3137,10 @@ def udf(
             If it is ``False``, attempting to register a UDF with a name that already exists
             results in a ``SnowparkSQLException`` exception being thrown. If it is ``True``,
             an existing UDF with the same name is overwritten.
+        if_not_exists: Whether to skip creation of a UDF when one with the same signature already exists.
+            The default is ``False``. ``if_not_exists`` and ``replace`` are used mutually exclusive
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/functions.py:3141]: omit "used".

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -3300,6 +3308,10 @@ def udtf(
             If it is ``False``, attempting to register a UDTF with a name that already exists
             results in a ``SnowparkSQLException`` exception being thrown. If it is ``True``,
             an existing UDTF with the same name is overwritten.
+        if_not_exists: Whether to skip creation of a UDTF when one with the same signature already exists.
+            The default is ``False``. ``if_not_exists`` and ``replace`` are used mutually exclusive
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/functions.py:3312]: remove "used"

File: src/snowflake/snowpark/udf.py [hunk 4]
```diff
@@ -521,6 +535,10 @@ def register(
                 If it is ``False``, attempting to register a UDF with a name that already exists
                 results in a ``SnowparkSQLException`` exception being thrown. If it is ``True``,
                 an existing UDF with the same name is overwritten.
+            if_not_exists: Whether to skip creation of a UDF when one with the same signature already exists.
+                The default is ``False``. ``if_not_exists`` and ``replace`` are used mutually exclusive
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/udf.py:539]: omit "used"

File: src/snowflake/snowpark/udf.py [hunk 5]
```diff
@@ -652,6 +672,10 @@ def register_from_file(
...
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/udf.py:676]: omit "used"

File: src/snowflake/snowpark/udtf.py [hunk 6]
```diff
@@ -369,6 +370,10 @@ def register(
                 If it is ``False``, attempting to register a UDTF with a name that already exists
                 results in a ``SnowparkSQLException`` exception being thrown. If it is ``True``,
                 an existing UDTF with the same name is overwritten.
+            if_not_exists: Whether to skip creation of a UDTF when one with the same signature already exists.
+                The default is ``False``. ``if_not_exists`` and ``replace`` are used mutually exclusive
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/udtf.py:374]: omit "used"

File: src/snowflake/snowpark/udtf.py [hunk 7]
```diff
@@ -479,6 +486,10 @@ def register_from_file(
...
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/udtf.py:490]: omit "used"

**Review by sfc-gh-mabrennan — COMMENTED** [review]
...just a few comments about the strings.

File: src/snowflake/snowpark/functions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/functions.py]: done

**Pull request merged**